docker/qchem.dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM ubuntu:20.04 AS compile-image
+
+# Setup and install Basic packages
+RUN apt-get update && apt-get install -y --no-install-recommends
+RUN DEBIAN_FRONTEND="noninteractive" apt-get install tzdata
+RUN apt-get install -y build-essential \
+        tzdata \
+        ca-certificates \
+        ccache \
+        cmake \
+        curl \
+	      git \
+	      python3 \
+        python3-pip \
+        python3-venv \
+        libjpeg-dev \
+        openbabel \
+        libpng-dev && \
+    rm -rf /var/lib/apt/lists/*
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache && ccache --set-config=cache_dir=/opt/ccache
+
+# Create and activate VirtualENV
+RUN python3 -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+#Setup and Build pennylane
+WORKDIR /opt/pennylane
+COPY  . .
+RUN git submodule update --init --recursive
+RUN  pip install wheel && pip install -r requirements.txt \
+        && python3 setup.py install \
+        && pip install jax jaxlib \
+        && pip install pytest pytest-cov pytest-mock flaky \
+        && make test
+
+# Setup and Build Qchem
+WORKDIR /opt/pennylane/qchem
+COPY  . .
+RUN git submodule update --init --recursive
+RUN  pip install wheel && pip install psi4 psi4-rt python=python3 -c psi4 && pip install -r requirements.txt \
```

Comment:
This command will fail, as Psi4 isn't available as a pip installable package. 
I'm currently investigating whether we can just use pyscf instead of psi4 here, as psi4 is conda-only, or build-from-source.

If PySCF isn't a good replacement, I'll help with setting up a build script for psi4, or we can use conda for this one image.

Reply:
I tried with "pip install openfermionpsi4" but build is still failing.

Reply:
Yes, openfermionpsi4 expects Psi4 to be installed separately, which is unfortunate, as this causes some challenges (as you have just seen :slightly_smiling_face: ). Psi4 also brings a hard requirement of Python3.7 with it. I'm currently chatting with members internally about the best approach here, and will update you as soon as we have a way forward.

Reply:
Thanks for the patience on this. So a quick follow-up: it seems that we can remove psi4 from the qchem image as it is an optional package (as per https://github.com/PennyLaneAI/pennylane/tree/master/qchem), and opt instead for PySCF (which can be pip installed), and `openfermion-pyscf` should allow the same functionalities as Psi4 and openfermion-psi4. Rather than installing directly from the qchem `requirements.txt` I think it would be better to be explicit about which packages to install: namely, install all packages but psi4 and openfermionpsi4. This should allow the image to be built.

Reply:
Yes, Tried that but still having issues while running "make test". it seems issue is with ENV path, will dig it tonight.